lib/apiservers/engine/backends/system.go

```diff
@@ -242,7 +245,77 @@ func (s *System) UnsubscribeFromEvents(chan interface{}) {
 
 func (s *System) AuthenticateToRegistry(ctx context.Context, authConfig *types.AuthConfig) (string, string, error) {
 	defer trace.End(trace.Begin(""))
-	return RegistryService.Auth(authConfig, dockerversion.DockerUserAgent(ctx))
+
+	fetcher := urlfetcher.NewURLFetcher(urlfetcher.FetcherOptions{
+		Timeout:  loginTimeout,
+		Username: authConfig.Username,
+		Password: authConfig.Password,
+	})
+
+	// Only look at V2 registries
+	var registryAddress string
+	if strings.Contains(authConfig.ServerAddress, "/v2") {
+		registryAddress = authConfig.ServerAddress
+	} else {
+		registryAddress = authConfig.ServerAddress + "/v2/"
```

Comment:
Would `strings.HasSuffix` be more suitable? Not that it makes a difference in this case, I think.